### PR TITLE
Improve MongoDB with Panache guide

### LINKED ISSUE #16399: Panache @BsonIgnore for methods
## Describe the bug
Since Panache is more Domain Driven, I added also some methods with small logic which like convert a different input parameter e.g. Date to LocalDateTime or get max Date from an Array... 
But for some reason `@BsonIgnore` only works on additional methods which are not setting fields.

### Expected behaviour
I expect that the method which I mark with `@BsonIgnore` gets actually ignored

### Actual behaviour
Example of a method which updates  my `lastActivity` (`LocalDateTime`) by the max parameter of an `LocalDateTime[]` (Array)
```
//Additional Method
public User setLastActivity(final LocalDateTime... lastActivities) {
        this.lastActivity = Arrays.stream(lastActivities).max(LocalDateTime::compareTo).orElse(getTimeLastYear());
        return this;
}

//Normal Method
public User setLastActivity(LocalDateTime lastActivity) {
        this.lastActivity = lastActivity;
        return this;
}
```
Error on save or get:
 
```
org.bson.codecs.configuration.CodecConfigurationException: Property 'lastActivity' in User, has differing data types: TypeData{type=LocalDateTime[]} and TypeData{type=LocalDateTime}.
```
Error on save or get when renaming the method:
```
Unable to get value for property 'lastActivities' in User

A custom Codec or PojoCodec may need to be explicitly configured and registered to handle this type.
```

## To Reproduce
* Create a method to update manually a field on the object with different input parameter - in my example an array
* Add `@BsonIgnore` to it
* Error is happening when you try to save or get the object

### PR DESCRIPTION
**Pull request by loicmathieu**
- Update link to the latest driver version
- Improve wording
- Add a section about the PojoCodecProvider

Fixes #16399

### CODE REVIEW

File: docs/src/main/asciidoc/mongodb-panache.adoc [hunk 1]
```diff
@@ -654,6 +654,21 @@ This can be achieved by setting to DEBUG the following log category inside your
 quarkus.log.category."io.quarkus.mongodb.panache.runtime".level=DEBUG
 ----
 
+== The PojoCodecProvider : easy object to BSON document conversion.
```
Comment (gsmet) [inline docs/src/main/asciidoc/mongodb-panache.adoc:657]:
```suggestion
== The PojoCodecProvider: easy object to BSON document conversion.
```

File: docs/src/main/asciidoc/mongodb-panache.adoc [hunk 2]
```diff
@@ -654,6 +654,21 @@ This can be achieved by setting to DEBUG the following log category inside your
...
+
+MongoDB with Panache uses the link:{mongodb-doc-root-url}/bson/pojos[PojoCodecProvider], with link:{mongodb-doc-root-url}/pojos/#pojo-support[automatic POJO support,
+to automatically convert your object to a BSON document.
+
+In case you encounter the `org.bson.codecs.configuration.CodecConfigurationException` exception, it means the codec is not able to
+automatically convert your object.
+This codec obeys the Java Bean standard, so it will successfully convert a POJO using public fields or getters/setters.
+You can use `@BsonIgnore` to make a field, or a getter/setter, ignored by the codec.
+
+If you class didn't obey these rules (for example by including a method that starts with `get` but is not a setter),
```
Comment (gsmet) [inline docs/src/main/asciidoc/mongodb-panache.adoc:667]:
```suggestion
If your class doesn't obey these rules (for example by including a method that starts with `get` but is not a setter),
```

**Review by gsmet — CHANGES_REQUESTED** [review]

**Comment by loicmathieu** [comment]
@gsmet thanks for your review, I applied your suggested changes.

**Comment by loicmathieu** [comment]
@gsmet can we merge this one ?

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Yes :).

**Pull request merged**